### PR TITLE
IX: Set bidVideo when category and duration is available

### PR DESCRIPTION
**Pull request by ixjohnny**
- Update IX adapter to set bidVideo if category and duration is available from response for video supply.

### CODE REVIEW

File: adapters/ix/ix.go [hunk 1]
```diff
@@ -402,9 +402,23 @@ func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReque
...
 				errs = append(errs, fmt.Errorf("Unmatched impression id: %s.", bid.ImpID))
 			}
+
+			var bidExtVideo *openrtb_ext.ExtBidPrebidVideo
+			var bidExt openrtb_ext.ExtBid
+			if bidType == openrtb_ext.BidTypeVideo {
+				unmarshalExtErr := json.Unmarshal(bid.Ext, &bidExt)
+				if unmarshalExtErr == nil && bidExt.Prebid != nil && bidExt.Prebid.Video != nil {
+					bidExtVideo = &openrtb_ext.ExtBidPrebidVideo{
+						Duration:        bidExt.Prebid.Video.Duration,
```
Comment (grossjo) [inline adapters/ix/ix.go:412]: can these be nil?

File: adapters/ix/ix.go [hunk 2]
```diff
@@ -402,9 +402,23 @@ func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReque
 			if !ok {
 				errs = append(errs, fmt.Errorf("Unmatched impression id: %s.", bid.ImpID))
 			}
+
+			var bidExtVideo *openrtb_ext.ExtBidPrebidVideo
+			var bidExt openrtb_ext.ExtBid
+			if bidType == openrtb_ext.BidTypeVideo {
+				unmarshalExtErr := json.Unmarshal(bid.Ext, &bidExt)
```
Comment (grossjo) [inline adapters/ix/ix.go:409]: Do we need to log on errors?

File: adapters/ix/ix.go [hunk 3]
```diff
@@ -402,9 +402,23 @@ func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReque
...
+					}
+				}
+			}
+
 			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
-				Bid:     &bid,
-				BidType: bidType,
+				Bid:      &bid,
+				BidType:  bidType,
+				BidVideo: bidExtVideo,
```
Comment (grossjo) [inline adapters/ix/ix.go:421]: could be unassigned.

File: adapters/ix/ix.go (hunk 1 above)
Comment (ixjohnny) [reply adapters/ix/ix.go:412]: PrimaryCategory/Duration? no, they will default to their zero value if they're not set.

File: adapters/ix/ix.go (hunk 3 above)
Comment (ixjohnny) [reply adapters/ix/ix.go:421]: It'll be nil in that case.

**Review by mlb7687 — APPROVED** [review]
Change LGTM

File: adapters/ix/ix.go (hunk 2 above)
Comment (ixjohnny) [reply adapters/ix/ix.go:409]: Done.

**Comment by VeronikaSolovei9** [comment]
LGTM

**Comment by ixjohnny** [comment]
Thanks for the review @VeronikaSolovei9 ! Is there anything else you need from me to approve the PR?

**Review by bsardo — APPROVED** [review]
LGTM. Just for awareness, we usually recommend using the JSON test framework to achieve adapter test coverage but that's not necessary here because our test framework doesn't currently support the `video` field on bid responses. We will add support in the future.

**Review by VeronikaSolovei9 — APPROVED** [review]

**Pull request merged**